### PR TITLE
Hatch 430/document text

### PR DESCRIPTION
**Pull request by CMLabay**
# Description

# Jira ticket links

# Test Plan

# Definition of done

- [ ] Does new code have 90% testing coverage (100% for `core`) of both unit and E2E tests?
- [ ] Is code secure? If applicable, add security notes to the description.
- [ ] Do all new TODO comments have Jira links with enough information?
- [ ] Were all changes published and deployed to [the grid](https://github.com/bitovi/hatchify-grid-demo)?
- [ ] Has the browser error-console been reviewed to not contain new errors introduced by these code changes?
- [ ] The site looks “good” above 1000px width.
- [ ] The site looks “good” when the window height is small. No double scrollbars.
- [ ] Were the changes tested to ensure 508 compliance?

### CODE REVIEW

File: doc/schema/attribute-types/text.md [hunk 1]
```diff
@@ -13,43 +13,69 @@ export const Todo: PartialSchema = {
 
 ## Parameters
 
-- `default` [{String}] - The default value of the attribute.  Example: `text({default: "hello"})`
-- `required` [{Boolean=false}] - If the attribute must be provided.
+|      key      | description                                                                 | type            | optional  | default |
+| ------------- | --------------------------------------------------------------------------- | :-------------: | :-------: | :-----: |
+| `default`     | The default value of the attribute. <br/> Example: `text({default: true})`  | `Boolean`       | Yes       |         |
+| `required`    | If the attribute must be provided.  <br/> Example: `text({required: true})` | `Boolean`       | Yes       | `false` |
```
Comment (arthurpankiewicz) [inline doc/schema/attribute-types/text.md:19]: the base data type used by text also has primary & unique - not sure if we should document them here or point to a table that has all the shared parameters

File: doc/schema/attribute-types/text.md [hunk 2]
```diff
@@ -13,43 +13,69 @@ export const Todo: PartialSchema = {
...
-<summary>
+### Querying Data
 
-## Advanced Details
+For text, use any string value in your queries as follows:
 
-</summary>
+```js
+GET /todos?filter[notes][$eq]=this%20is%20super%20important //all todos with notes that equal "this is super important"
+GET /todos?filter[notes][$eq]=null //all todos with notes that are null
```
Comment (arthurpankiewicz) [inline doc/schema/attribute-types/text.md:33]: not sure if we should document this, or the future behaviour and put a 🛑 emoji. this iteration will return `notes` that are `null` or `"null"`?

File: doc/schema/attribute-types/text.md [hunk 3]
```diff
@@ -13,43 +13,69 @@ export const Todo: PartialSchema = {
...
+```ts
+Todo.createOne({ attributes: { notes: "this is super important" } })
+
+const [todo, todoMeta] = hatchedReactRest.Todo.useOne({ id })
+todo.notes //-> string or null
 ```
-  
-</details>
 
+## Grid Behavior
```
Comment (arthurpankiewicz) [inline doc/schema/attribute-types/text.md:73]: we don't have anything in the frontend code for this now, but what is the expected behaviour for `reallyrealy really really long` text values? should we do an ellipsis at a certain length? (does mui already handle this?)

we should document expected future behaviour with a 🛑

File: doc/schema/attribute-types/text.md (hunk 1 above)
Comment (CMLabay) [reply doc/schema/attribute-types/text.md:19]: I feel like they should be here if the other two are. I'll add them

File: doc/schema/attribute-types/text.md (hunk 3 above)
Comment (CMLabay) [reply doc/schema/attribute-types/text.md:75]: It does not currently handle that. Asking in slack

**Review by arthurpankiewicz — APPROVED** [review]

File: doc/schema/attribute-types/text.md [hunk 4]
```diff
@@ -13,43 +13,72 @@ export const Todo: PartialSchema = {
...
 ## Parameters
 
-- `default` [{String}] - The default value of the attribute.  Example: `text({default: "hello"})`
-- `required` [{Boolean=false}] - If the attribute must be provided.
+|      key      | description                                                                      | type            | optional  | default |
+| ------------- | -------------------------------------------------------------------------------- | :-------------: | :-------: | :-----: |
+| `default`     | The default value of the attribute. <br/> Example: `text({default: 'no text'})`  | `String`        | Yes       |         |
+| `required`    | If the attribute must be provided.  <br/> Example: `text({required: true})`      | `Boolean`       | Yes       | `false` |
+| `primary`     | The attribute is a primary key.     <br/> Example: `text({primary: true})`       | `Boolean`       | Yes       | `false` |
+| `unique`      | The attribute must be unique.       <br/> Example: `text({unique: true})`        | `Boolean`       | Yes       | `false` |
```
[the remark this answers is not in the document]
Comment (nlundquist) [reply doc/schema/attribute-types/text.md:21]: yeah a param to support `tiny` could be added here. imo no real value to documenting a "nice to have" that isn't on a firm roadmap yet.

we might always end up changing our minds and implementing `tiny` as it's own type.

**Pull request merged**